src/main/java/microsoft/exchange/webservices/data/AutodiscoverService.java

```diff
@@ -1458,91 +1464,89 @@ else if (this.getCredentials()instanceof X509CertificateCredentials)
    * @throws Exception the exception
    */
   private boolean tryGetEnabledEndpointsForHost(String host,
-      OutParam<EnumSet<AutodiscoverEndpoints>> endpoints)
-      throws Exception {
+      OutParam<EnumSet<AutodiscoverEndpoints>> endpoints) throws Exception {
     this.traceMessage(TraceFlags.AutodiscoverConfiguration, String.format(
         "Determining which endpoints are enabled for host %s", host));
 
-    // We may get redirected to another host. And therefore need to limit
-    // the number
-    // of redirections we'll tolerate.
+    // We may get redirected to another host. And therefore need to limit the number of redirections we'll
+    // tolerate.
     for (int currentHop = 0; currentHop < AutodiscoverMaxRedirections; currentHop++) {
-      URI autoDiscoverUrl = new URI(String.format(
-          AutodiscoverLegacyHttpsUrl, host));
+      URI autoDiscoverUrl = new URI(String.format(AutodiscoverLegacyHttpsUrl, host));
 
       endpoints.setParam(EnumSet.of(AutodiscoverEndpoints.None));
 
-      HttpWebRequest request = new HttpClientWebRequest(this.getSimpleHttpConnectionManager());
+      HttpWebRequest request = null;
```

Comment:
I'm not very comfortable with refactoring these methods myself. Perhaps we open an issue or address it in a separate PR.